client/src/contexts/AuthContext.js

```diff
@@ -0,0 +1,44 @@
+import React, { createContext, useState } from "react";
+import PropTypes from "prop-types";
+import axios from "axios";
+
+export const AuthContext = createContext();
+
+AuthContextProvider.propTypes = {
+  children: PropTypes.node,
+};
+
+export function AuthContextProvider({ children }) {
+  const [token, setToken] = useState();
+  const [user, setUser] = useState();
+
+  const loginWithCode = (code) => {
+    if (token) {
+      return;
+    }
+    axios
+      .post("/login/oauth/github/token", { code })
+      .then((res) => res.data.access_token)
+      .then((token) => {
+        setToken(token);
+        return axios.get("https://api.github.com/user", {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        });
+      })
+      .then((res) => setUser(res.data))
+      .catch(console.log);
```

Comment:
```suggestion
      .catch(console.error);
```

This makes it possible to filter your console and expand/collapse error stack.

But I would love to see this error in a state which is exposed to the context consumer.